### PR TITLE
Audio file format plugins: set channels information

### PR DESCRIPTION
**Pull request by iDarkTemplar**

### CODE REVIEW

**Comment by iDarkTemplar** [comment]
This change depends on https://github.com/audacious-media-player/audacious/pull/38 to compile and work.

I've tested this change only on .mp3 files with 2 channels and .flac files with 1 channel and 2 channels, but most supported file types should display proper channels count as well, except for .cue files.

To display channels playlist has to be refreshed

File: src/amidiplug/amidi-plug.cc [hunk 1]
```diff
@@ -150,15 +154,11 @@ bool AMIDIPlug::read_tag (const char * filename, VFSFile & file, Tuple & tuple,
 
     tuple.set_str (Tuple::Codec, "MIDI");
     tuple.set_int (Tuple::Length, mf.length / 1000);
+    tuple.set_int (Tuple::Channels, s_channels);
```
Comment (jlindgren90) [inline src/amidiplug/amidi-plug.cc:157]: read_tag() should not be using a static variable; it needs to be thread-safe.

File: src/ffaudio/ffaudio-core.cc [hunk 2]
```diff
@@ -431,8 +431,13 @@ bool FFaudio::read_tag (const char * filename, VFSFile & file, Tuple & tuple, In
     if (! find_codec (ic.get (), & cinfo))
         return false;
 
+    ScopedContext context (cinfo);
+    if (LOG (avcodec_open2, context.ptr, cinfo.codec, nullptr) < 0)
```
Comment (jlindgren90) [inline src/ffaudio/ffaudio-core.cc:435]: Not a fan of doing a stream-open just to get the channel count

File: src/amidiplug/amidi-plug.cc (hunk 1 above)
Comment (iDarkTemplar) [reply src/amidiplug/amidi-plug.cc:157]: Would explicit call to function **backend_audio_info** be fine?

File: src/ffaudio/ffaudio-core.cc (hunk 2 above)
Comment (iDarkTemplar) [reply src/ffaudio/ffaudio-core.cc:435]: I don't see any other way to get information about channels here, but I'm not familiar with ffmpeg API. As alternative, changes for this file may be dropped.

File: src/amidiplug/amidi-plug.cc (hunk 1 above)
Comment (jlindgren90) [reply src/amidiplug/amidi-plug.cc]: That seems fine.

File: src/ffaudio/ffaudio-core.cc (hunk 2 above)
Comment (jlindgren90) [reply src/ffaudio/ffaudio-core.cc:435]: Let's do that for now, just leave this one out.

**Comment by iDarkTemplar** [comment]
Is anything still blocking this change?

**Pull request merged**